src/wasm2obj.rs

```diff
@@ -87,18 +93,39 @@ fn main() {
     }
 }
 
-fn handle_module(path: PathBuf, output: &str) -> Result<(), String> {
+fn handle_module(path: PathBuf, target: &Option<String>, output: &str)  -> Result<(), String> {
     let data = match read_wasm_file(path) {
         Ok(data) => data,
         Err(err) => {
             return Err(String::from(err.description()));
         }
     };
 
-    // FIXME: Make the target a parameter.
-    let (flag_builder, isa_builder) = cranelift_native::builders().unwrap_or_else(|_| {
-        panic!("host machine is not a supported target");
-    });
+    let (flag_builder, isa_builder) = match target {
+        Some(target) => {
+            let target = Triple::from_str(&target).map_err(|_| "could not parse --target")?;
+            let mut flag_builder = settings::builder();
+            match target.operating_system {
+                OperatingSystem::Windows => {
+                    flag_builder.set("call_conv", "windows_fastcall").unwrap();
+                },
+                _ => {
+                    flag_builder.set("call_conv", "system_v").unwrap();
+                }
+            };
```

Comment:
Hmm. This actually tells me that "call_conv" shouldn't be a global setting in Cranelift. We should really just have cranelift-wasm get the default calling convention from the `Triple`. I filed https://github.com/CraneStation/cranelift/issues/555 to track this. This code in wasmtime is ok for the time being though.

Reply:
I assume it would still be desirable to have some way of overriding the calling convention for some subset of functions from wasmtime, though?  I suppose that's way down the line, though. :smile:

Reply:
Yes, this setting is just controlling the "default" calling convention. Cranelift supports using different calling conventions or use custom ones on a per-function basis.